ethexe/runtime/common/src/lib.rs

```diff
@@ -86,40 +90,43 @@ impl<S: Storage> TransitionController<'_, S> {
 
         let res = f(&mut state, self.storage, self.transitions);
 
+        let queue_size = state.queue.cached_queue_size;
         let new_state_hash = self.storage.write_state(state);
 
-        self.transitions.modify_state(program_id, new_state_hash);
+        self.transitions
+            .modify_state(program_id, new_state_hash, queue_size);
 
         res
     }
 }
 
-pub fn process_next_message<S, RI>(
+pub fn process_queue<S, RI>(
```

Comment:
TODO for addition of queue/gas limit for processing?

Reply:
or maybe even handle it now, so previously updated tests that use process_next_message would keep untouched, since process_next_message will be an equivalent to process_queue(msg_limit: 1)

Reply:
Added TODO, also this https://github.com/gear-tech/gear/pull/4488#issuecomment-2818445693

Reply:
Implemented in https://github.com/gear-tech/gear/pull/4646